pkg/controller/podgc/gc_controller.go

```diff
@@ -124,6 +130,57 @@ func isPodTerminated(pod *v1.Pod) bool {
 	return false
 }
 
+// isPodTerminating returns true if the pod is terminating.
+func isPodTerminating(pod *v1.Pod) bool {
+	if pod.ObjectMeta.DeletionTimestamp != nil {
+		klog.Infof("garbage collecting pod %s that is terminating. Phase [%v]", pod.Name, pod.Status.Phase)
```

Comment:
Conventionally, an `isFoo` function is pretty trivial and "utility" - logging from within here seems surprising.  I think you should log this at the caller.

Reply:
Also - check the logs - there's ANOTHER log info below - do you need both?